plugins/importexport/native/filter/NativeXmlPublicationFilter.inc.php

```diff
@@ -0,0 +1,272 @@
+<?php
+
+/**
+ * @file plugins/importexport/native/filter/NativeXmlPublicationFilter.inc.php
+ *
+ * Copyright (c) 2014-2020 Simon Fraser University
+ * Copyright (c) 2000-2020 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class NativeXmlPublicationFilter
+ * @ingroup plugins_importexport_native
+ *
+ * @brief Class that converts a Native XML document to a set of articles.
+ */
+
+import('lib.pkp.plugins.importexport.native.filter.NativeXmlPKPPublicationFilter');
+
+class NativeXmlPublicationFilter extends NativeXmlPKPPublicationFilter {
+	/**
+	 * Constructor
+	 * @param $filterGroup FilterGroup
+	 */
+	function __construct($filterGroup) {
+		parent::__construct($filterGroup);
+	}
+
+
+	//
+	// Implement template methods from PersistableFilter
+	//
+	/**
+	 * @copydoc PersistableFilter::getClassName()
+	 */
+	function getClassName() {
+		return 'plugins.importexport.native.filter.NativeXmlPublicationFilter';
+	}
+
+	/**
+	 * Get the method name for inserting a published submission.
+	 * @return string
+	 */
+	function getPublishedSubmissionInsertMethod() {
+		return 'insertObject';
+	}
+
+	/**
+	 * Handle an Article import.
+	 * The Article must have a valid section in order to be imported
+	 * @param $node DOMElement
+	 */
+	function handleElement($node) {
+		$deployment = $this->getDeployment();
+		$context = $deployment->getContext();
+		$sectionAbbrev = $node->getAttribute('section_ref');
+		if ($sectionAbbrev !== '') {
+			$sectionDao = DAORegistry::getDAO('SectionDAO'); /* @var $sectionDao SectionDAO */
+			$section = $sectionDao->getByAbbrev($sectionAbbrev, $context->getId());
+			if (!$section) {
+				$deployment->addError(ASSOC_TYPE_SUBMISSION, NULL, __('plugins.importexport.native.error.unknownSection', array('param' => $sectionAbbrev)));
+			} else {
+				return parent::handleElement($node);
+			}
+		}
+	}
+
+	/**
+	 * @see Filter::process()
+	 * @param $document DOMDocument|string
+	 * @return array Array of imported documents
+	 */
+	function &process(&$document) {
```

Comment:
I think this can be removed entirely as well.

Reply:
Fixed.